pandas/core/reshape/merge.py

```diff
@@ -1452,6 +1454,147 @@ def _validate(self, validate: str) -> None:
             raise ValueError("Not a valid argument for validate")
 
 
+def _chunks(lst, n):
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+_DEFAULT_LEFT_CHUNK_SIZE = _DEFAULT_RIGHT_CHUNK_SIZE = 1000
+
+
+# TODO: perform conditional merge as optimized cython,
+#       rather than chunked/filtered cross merge, also
+#       incorporate v2 conditional merge logic into _MergeOperation,
+#       rather than the subclass below
+class _ConditionalEnabledMergeOperation(_MergeOperation):
+    def __init__(
+        self,
+        left: DataFrame | Series,
+        right: DataFrame | Series,
+        how: str = "inner",
+        on: IndexLabel | Callable | None = None,
+        left_on: IndexLabel | None = None,
+        right_on: IndexLabel | None = None,
+        axis: int = 1,
+        left_index: bool = False,
+        right_index: bool = False,
+        sort: bool = True,
+        suffixes: Suffixes = ("_x", "_y"),
+        copy: bool = True,
+        indicator: bool = False,
+        validate: str | None = None,
+        left_chunk_size: int | None = None,
+        right_chunk_size: int | None = None,
+    ):
+        """
+        Note: Callable `on` function should respect future desired interface.
+        Currently the function "could" make use of df specific logic
+        to return a boolean mask (e.g. left.time.dt.date == right.time.dt.date),
+        but eventually when this implementation is replaced we might not have
+        that ability. Should we (1) restrict those cases now? (2) allow them and
+        fail in the future? or (3) plan to develop future implementation to work
+        with these constructs as well?
+        """
+        self.is_conditional_merge = False
+        if callable(on):
```

Comment:
This should be in `_MergeOperation._validate_specification`

Reply:
updated, separated it further into `_validate_conditional_merge_specification` which is called in your stated method